includes/course-theme/class-sensei-course-theme-editor.php

```diff
@@ -0,0 +1,372 @@
+		$templates = array_filter(
+			$templates,
+			function( $template ) {
+				return ! preg_match( '#^' . Sensei_Course_Theme::THEME_NAME . '//.*$#', $template->id );
+			}
+		);
+
+		$theme_templates = array_values( $this->get_block_templates() );
+
+		$post_type = $query['post_type'] ?? null;
+
+		if ( $post_type ) {
+			$theme_templates = array_filter(
+				$theme_templates,
+				function( $template ) use ( $post_type ) {
+					return in_array( $post_type, $template->post_types, true );
+				}
+			);
+		}
+
+		return array_merge( $theme_templates, $templates );
+
+	}
+
+	/**
+	 * Get block templates, including user-customized overrides.
+	 *
+	 * @return array
+	 */
+	public function get_block_templates() {
+
+		$this->load_file_templates();
+
+		$db_templates = $this->get_custom_templates();
+		$templates    = [];
+
+		foreach ( $this->file_templates as $name => $template ) {
+			$db_template     = $db_templates[ $name ] ?? null;
+			$template_object = (object) $template;
+
+			if ( ! empty( $db_template ) ) {
+				$template_object = $this->build_template_from_post( $db_template );
+			} else {
+				$template_object->wp_id  = null;
+				$template_object->author = null;
+			}
+			$templates[ $name ] = $template_object;
+		}
+
+		return $templates;
+	}
+
+	/**
+	 * Get templates customized by the user.
+	 *
+	 * @return array
+	 */
+	private function get_custom_templates() {
+
+		$db_templates_query = new \WP_Query(
+			[
+				'post_type'      => 'wp_template',
+				'posts_per_page' => -1,
+				'no_found_rows'  => true,
+				'tax_query'      => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+					[
+						'taxonomy' => 'wp_theme',
+						'field'    => 'name',
+						'terms'    => [ self::THEME_PREFIX ],
+					],
+				],
+			]
+		);
+
+		$db_templates = $db_templates_query->posts ?? [];
+
+		return array_column( $db_templates, null, 'post_name' );
+	}
+
+	/**
+	 * Adds the Appearance -> Editor menu item, unless it's already there because the active theme is a block theme.
+	 *
+	 * @access private
+	 */
+	public function add_admin_menu_site_editor_item() {
+
+		if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+			return;
+		}
+
+		add_theme_page(
+			__( 'Editor', 'sensei-lms' ),
+			sprintf(
+			/* translators: %s: "beta" label */
+				__( 'Editor %s', 'sensei-lms' ),
+				'<span class="awaiting-mod">' . __( 'beta', 'sensei-lms' ) . '</span>'
+			),
+			'edit_theme_options',
+			'site-editor.php?postType=wp_template'
+		);
+
+	}
+
+
+	/**
+	 * Add template editing hooks for site editor and related API requests.
+	 *
+	 * @access private
+	 */
+	public function maybe_add_site_editor_hooks() {
+
+		$uri = isset( $_SERVER['REQUEST_URI'] ) ? esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
+
+		$is_site_editor      = preg_match( '#/wp-admin/site-editor.php#i', $uri );
```

Comment:
Can't really filter for that, because the Site Editor works as a single page app, the page is not reloaded when opening a template to edit or going back to the templates list. 
But we are planning to get the blocks usable in other theme templates too so it should be fine.